### PR TITLE
Steps in standard flow are now gated based on the existence of their …

### LINKED ISSUE #113: Add a pipeline builder to generate test pipeline based on scenario artiacts
The files within each scenario should tell the reporting code what shape the resulting pipeline should be. This should modify both what code is run for that scenario and what shape is persisted to the final report.

For example:
If there is ONLY a `code.zip` file in the scenario, the resulting pipeline would include the `code_to_amr` process but would not include any other steps because the other steps depend on a `paper.pdf`.

### PR DESCRIPTION
**Pull request by Sorrento110**
…prerequisites. Added dataset uploading to seeding.

# Description

* Adds support for datasets and equations in the testing pipeline.
* The pipeline will now report on all pieces, and will flag some of them as non-applicable if the scenario shouldn't run them.

Resolves #113 #104 #103

### CODE REVIEW

File: reporting/scenarios/SIDARTHE Equations/description.txt
Comment (brandomr) [inline reporting/scenarios/SIDARTHE Equations/description.txt:1]: This isn't an appropriate description of the test scenario. It should be something like: "Testing SIDARTHE based on equations curated from its paper"

File: reporting/scenarios/SIDARTHE Code/description.txt
Comment (brandomr) [inline reporting/scenarios/SIDARTHE Code/description.txt:1]: Same as the `SIDARTHE Equations` scenario--this shouldn't be just the abstract but should actually describe the scenario: testing SIDARTHE based on conversion of its code to an AMR

File: reporting/tests/report.py [hunk 1]
```diff
@@ -120,88 +119,159 @@ def run_km_job(url, scenario, task_name):
...
-        "time" : execution_time,
-        "accuracy" : None,
-        "success" : success
-    })
+    result.update({"time": execution_time, "accuracy": None, "success": success})
     return result
 
 
+def non_applicable_run(task_name):
+    return (task_name, {"success": None, "time": 0, "accuracy": None})
```
Comment (brandomr) [inline reporting/tests/report.py:129]: Should `time` also be `None`?

**Review by brandomr — CHANGES_REQUESTED** [review]
I feel like one of the nice aspects of our previous version of tests was we could surface the logs for each task in the Dashboard. Can we add the logging to the generated report here--or should that be a separate issue due to level of effort?

File: reporting/scenarios/SIDARTHE Equations/description.txt
Comment (Sorrento110) [reply reporting/scenarios/SIDARTHE Equations/description.txt:1]: Alright I will change them! These were just the descriptions in the old scenario.

File: reporting/tests/report.py (hunk 1 above)
Comment (Sorrento110) [reply reporting/tests/report.py:129]: @fivegrant we are keeping time as a number for sake of compatibility with integration dashboard? Should we change this to `None` so no runtime shows up?

Comment (fivegrant) [reply reporting/tests/report.py]: https://github.com/DARPA-ASKEM/integration-dashboard/commit/ea98617bef8c5c618699ea9d3e957fec7cb23671
we can now handle nulls

Comment (Sorrento110) [reply reporting/tests/report.py]: Changed to `None` with changes to integration dashboard also happening to support this.

**Review by brandomr — APPROVED** [review]

**Pull request merged**